src/compiler/resolutionCache.ts

```diff
@@ -59,10 +63,15 @@ namespace ts {
         // The values are Map of resolutions with key being name lookedup.
         const resolvedModuleNames = createMap<Map<ResolvedModuleWithFailedLookupLocations>>();
         const perDirectoryResolvedModuleNames = createMap<Map<ResolvedModuleWithFailedLookupLocations>>();
+
         const resolvedTypeReferenceDirectives = createMap<Map<ResolvedTypeReferenceDirectiveWithFailedLookupLocations>>();
         const perDirectoryResolvedTypeReferenceDirectives = createMap<Map<ResolvedTypeReferenceDirectiveWithFailedLookupLocations>>();
+
         const getCurrentDirectory = memoize(() => resolutionHost.getCurrentDirectory());
 
+        const failedLookupDefaultExtensions = [Extension.Ts, Extension.Tsx, Extension.Js, Extension.Jsx, Extension.Json];

```

Comment:
* Why doesn't this include `.d.ts`?
* I think this should be defined next to `supportedTypeScriptExtensions` and the other extensions sets.

Reply:
Updated with note. But the reason is /**
         * These are the extensions that failed lookup files will have by default,
         * any other extension of failed lookup will be store that path in custom failed lookup path
         * This helps in not having to comb through all resolutions when files are added/removed
         * Note that .d.ts file also has .d.ts extension hence will be part of default extensions
         */  Also i think it makes sense to have this in the resolution cache instead of anywhere else

Reply:
In `Note that .d.ts file also has .d.ts` I think you mean `also has .ts`.
Also, it would be clearer to use the variable name `customFailedLookupPaths` instead of `custom failed lookup path`.